src/GRA.Domain.Service/AvatarService.cs

```diff
@@ -431,34 +444,37 @@ public async Task<ICollection<AvatarElement>> GetUserAvatarAsync()
             return await _avatarElementRepository.GetUserAvatarAsync(GetActiveUserId());
         }
 
-        public async Task<Dictionary<AvatarBundle, bool>> GetUserUnlockBundlesAsync()
+        public async Task<List<AvatarBundle>> GetUserUnlockBundlesAsync()
         {
             var userHistory = await _avatarBundleRepository.UserHistoryAsync(GetActiveUserId());
-            var bundles = new Dictionary<AvatarBundle, bool>();
+            var bundles = new List<AvatarBundle>();
             foreach (var historyItem in userHistory.Where(_ => _.AvatarBundleId.HasValue))
             {
-                if (!bundles.Keys.Any(bundle => bundle.Id == historyItem.AvatarBundleId))
+                if (!bundles.Any(bundle => bundle.Id == historyItem.AvatarBundleId))
                 {
-                    bundles.Add(await GetBundleByIdAsync(historyItem.AvatarBundleId.Value),
-                        historyItem.HasBeenViewed ?? false);
+                    var bundle = await GetBundleByIdAsync(historyItem.AvatarBundleId.Value);
+                    bundle.HasBeenViewed = historyItem.HasBeenViewed ?? false;
+                    bundles.Add(bundle);
                 }
             }
             return bundles;
         }
 
-        public async Task UpdateUserLogsAsync(List<UserLog> userLog)
+        public async Task UpdateBundleHasBeenViewedAsync(int bundleId)
         {
-            foreach (var bundle in userLog)
-            {
-                await _avatarBundleRepository.UpdateHasBeenViewedAsync(GetActiveUserId(), 
-                    bundle.AvatarBundleId.Value);
-            }
+            await _avatarBundleRepository.UpdateHasBeenViewedAsync(GetActiveUserId(),
+                bundleId);
         }
 
         public async Task UpdateUserAvatarAsync(ICollection<AvatarLayer> selectionLayers)
         {
             var activeUserId = GetActiveUserId();
-            var layers = await _avatarLayerRepository.GetAllAsync(GetCurrentSiteId());
+            var currentCultureName = _userContextProvider.GetCurrentCulture()?.Name;
+            var currentLanguageId = currentCultureName != null ?
+                await _languageService.GetLanguageIdAsync(currentCultureName) :
+                await _languageService.GetDefaultLanguageIdAsync();
+            VerifyManagementPermission();
```

Comment:
VerifyManagementPermission shouldn't be called here

Reply:
Fixed in commit 1e29452a573b621d3ecb1dfc99d6cf300a36f0b7